### PR TITLE
Generate obstacles on battlefield based on the battle tile index

### LINKED ISSUE #1294: Position of obstacles on the battlefield when a hero is attacked multiple times at the same position on the adventure map
In fheroes2, if a player attacks several times the same enemy hero located in the same location on the adventure map with several of his heroes, the obstacles present on the battlefield are not positioned in the same hexagons.

In the original game, the obstacles are always positioned in the same hexagons, even if the opposing hero has moved to other locations on the adventure map the following turn and then returned to the same original location on the adventure map.

In fheroes2:

Position of obstacles on the battlefield when a hero is attacked multiple times at the same position on the adventure map 01

Position of obstacles on the battlefield when a hero is attacked multiple times at the same position on the adventure map 02

Here is the test scenario file:
[Abandoned mine no victory window.zip](https://github.com/ihhub/fheroes2/files/5098267/Abandoned.mine.no.victory.window.zip)

Attack the green hero.

### PR DESCRIPTION
**Pull request by oleg-derevenetz**
fix #1294

### CODE REVIEW

File: src/fheroes2/battle/battle_arena.cpp [hunk 1]
```diff
@@ -286,12 +286,14 @@ Battle::Arena::Arena( Army & a1, Army & a2, s32 index, bool local )
     else
     // set obstacles
     {
-        icn_covr = Rand::Get( 0, 99 ) < 40 ? GetCovr( world.GetTiles( index ).GetGround() ) : ICN::UNKNOWN;
+        std::mt19937 seededGen( static_cast<uint32_t>( index ) );
```
[the remark this answers is not in the document]
Comment (ihhub) [reply src/fheroes2/battle/battle_arena.cpp:289]: I think we need to use `GetMapSeed()` from the world cobject plus add to it the index of the tile. What do you think?

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @oleg-derevenetz , I left just 2 comments regarding the implementation. Could you please check them?

File: src/fheroes2/battle/battle_arena.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (oleg-derevenetz) [reply src/fheroes2/battle/battle_arena.cpp:289]: Hi @ihhub

> Wouldn't we have the same obstacles at the same map indexes across all maps?

If they have the same terrain type, yes. Is this an issue?

> I think we need to use GetMapSeed() from the world cobject plus add to it the index of the tile. What do you think?

Yes, it's possible if the above is considered as an issue.

[the remark this answers is not in the document]
Comment (ihhub) [reply src/fheroes2/battle/battle_arena.cpp:289]: Hi @oleg-derevenetz , yes, it is the issue. Each map should have a unique battle obstacles so players can't exploit it :)

**Comment by ihhub** [comment]
@oleg-derevenetz , thank you so much for fixing this bug!

**Pull request merged**